### PR TITLE
SJQ2T-25: Validation BE

### PR DESCRIPTION
**Pull request by MarkoRadunovicHtec**
Validation implemented in UserRegisterRequest, adequate handler added into CustomExceptionHandler, phone number already in use validation implemented

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/skynest/exception/CustomException.java [hunk 1]
```diff
@@ -6,13 +6,13 @@
 
 @Getter
 @AllArgsConstructor
-public class UserException extends RuntimeException {
+public class CustomException extends RuntimeException {
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/exception/CustomException.java]: Because lots of PR already use the UserException, it is good practice this refactoring to be done in separate PR

File: backend/src/main/java/com/htecgroup/skynest/exception/CustomExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  @ExceptionHandler(value = {MethodArgumentNotValidException.class})
+  public ResponseEntity<ErrorMessage> handleValidationException(
+      MethodArgumentNotValidException ex) {
+    List<String> errors = new ArrayList<>();
+    ex.getBindingResult()
+        .getAllErrors()
+        .forEach(
+            error -> {
+              String errorMessage =
+                  ((FieldError) error).getField() + " " + error.getDefaultMessage();
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/exception/CustomExceptionHandler.java]: you can use String.format instead of this concatenation

File: backend/src/main/java/com/htecgroup/skynest/model/request/UserRegisterRequest.java [hunk 3]
```diff
@@ -1,14 +1,37 @@
...
+  @Email(message = "format is not valid", regexp = "(.+)@(\\S+)\\.(\\S+)")
+  @Size(max = 254, message = "length cannot be over 254 characters")
   private String email;
+
+  @Pattern(
+      regexp = "(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)[a-zA-Z\\d]{8,50}",
+      message = "format not valid")
   private String password;
+
+  @NotBlank(message = "cannot be empty")
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/model/request/UserRegisterRequest.java]: Can it be Null ? Applies to all required properties

File: backend/src/main/java/com/htecgroup/skynest/exception/CustomExceptionHandler.java (hunk 2 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/exception/CustomExceptionHandler.java]: Will do it.

File: backend/src/main/java/com/htecgroup/skynest/exception/CustomException.java (hunk 1 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/exception/CustomException.java]: Of corse. I ll redo changes. And than after it can be done in separate PR

File: backend/src/main/java/com/htecgroup/skynest/model/request/UserRegisterRequest.java (hunk 3 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/model/request/UserRegisterRequest.java]: @NotBlank forbid null values. So I can add that in message. For example Cannot be null or empty.

File: backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java [hunk 4]
```diff
@@ -2,20 +2,45 @@
...
+@Slf4j
 public class UserExceptionHandler {
 
   @ExceptionHandler(value = {UserException.class})
   public ResponseEntity<ErrorMessage> handleUserException(UserException ex, WebRequest webRequest) {
     ErrorMessage errorMessage =
         new ErrorMessage(
             ex.getMessage(), ex.getStatus().value(), DateTimeUtil.currentTimeFormatted());
-
+    log.error(ex.toString(), ex);
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java]: it is better to use log.error("Handle UserException. Error reason: ", ex) or something like that. The way that you used you would log the same error twice I believe

File: backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java [hunk 5]
```diff
@@ -2,20 +2,45 @@
...
+  public ResponseEntity<ErrorMessage> handleValidationException(
+      MethodArgumentNotValidException ex) {
+    List<String> errors = new ArrayList<>();
+    ex.getBindingResult()
+        .getAllErrors()
+        .forEach(
+            error ->
+                errors.add(
+                    String.format(
+                        "%s %s", ((FieldError) error).getField(), error.getDefaultMessage())));
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java]: Instead of forEach, take a look at **_map_** and _**collect**_ options and use them as combination.

File: backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java (hunk 4 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java]: I willl fix it

File: backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java (hunk 5 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java]: I'll take a look and use it.

**Review by petarstoimenovskihtec — APPROVED** [review]

**Comment by MarkoRadunovicHtec** [comment]
Validation for User register request implemented, Handler added for MethodArgumentNotValidException, ErrorMessage structure edited
